pkg/sql/crdb_internal.go

```diff
@@ -3937,9 +3936,13 @@ CREATE TABLE crdb_internal.gossip_nodes (
 		}
 
 		alive := make(map[roachpb.NodeID]tree.DBool)
+		now := timeutil.Now()
 		for _, d := range descriptors {
-			if _, err := g.GetInfo(gossip.MakeGossipClientsKey(d.NodeID)); err == nil {
-				alive[d.NodeID] = true
+			var gossipLiveness livenesspb.Liveness
+			if err := g.GetInfoProto(gossip.MakeNodeLivenessKey(d.NodeID), &gossipLiveness); err == nil {
```

Comment:
Was this change needed to pass some tests?

Reply:
I haven't actually tried running any tests without this. It just seemed like it would be pretty bad if `crdb_internal.gossip_nodes` suddenly switched to always saying that all nodes are not live (due to the gossip clients keys being missing). Especially since IIRC it's the table that's used by CLI commands like `cockroach node status`.

And basing it on node liveness seems like a reasonable alternative, although I'm open to other ideas.